airflow/providers/amazon/aws/hooks/redshift_data.py

```diff
@@ -201,3 +217,56 @@ def get_table_primary_key(
                 break
 
         return pk_columns or None
+
+    async def check_query_is_finished_async(
+        self, statement_id: str, poll_interval: int = 10
+    ) -> dict[str, str]:
+        """Async function to check statement is finished.
+
+        It takes statement_id, makes async connection to redshift data to get the query status
+        by statement_id and returns the query status.
+
+        :param statement_id: the UUID of the statement
+        :param poll_interval: how often in seconds to check the query status
+        """
+        try:
+            client = await sync_to_async(self.get_conn)()
+            while await self.is_still_running(statement_id):
+                await asyncio.sleep(poll_interval)
+
+            resp = client.describe_statement(Id=statement_id)
+            status = resp["Status"]
+            if status == FINISHED_STATE:
+                return {"status": "success", "statement_id": statement_id}
+            elif status == FAILED_STATE:
+                return {
+                    "status": "error",
+                    "message": f"Error: {resp['QueryString']} query Failed due to, {resp['Error']}",
+                    "statement_id": statement_id,
+                    "type": status,
+                }
+            elif status == ABORTED_STATE:
+                return {
+                    "status": "error",
+                    "message": "The query run was stopped by the user.",
+                    "statement_id": statement_id,
+                    "type": status,
+                }
+
+            return {
+                "status": "error",
+                "message": f"Unexpected statue {status}",
+                "statement_id": statement_id,
+                "type": status,
+            }
+        except botocore.exceptions.ClientError as error:
+            return {"status": "error", "message": str(error), "type": "ERROR"}
+
+    async def is_still_running(self, statement_id: str) -> bool:
+        """Async function to check whether the query is still running.
+
+        :param statement_id: the UUID of the statement
+        """
+        client = await sync_to_async(self.get_conn)()
```

Comment:
Is `client` here an asynchronous or synchronous? Are you able to use the `async_conn` that is defined in the aws base hook?

Reply:
Yep, just tested. It works fine! Thanks for your suggestion!